src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -0,0 +1,311 @@
+            ParentProvider?.GetResource().Attributes.ToDictionary() ?? new Dictionary<string, object>(0));
+
+        hub.ConfigureScope(scope => scope.AddTransactionProcessor(TransactionProcessor));
+    }
+
+    internal static Transaction TransactionProcessor(Transaction transaction)
+    {
+        var activity = Activity.Current;
+        if (activity != null)
+        {
+            var trace = transaction.Contexts.Trace;
+            trace.TraceId = activity.TraceId.AsSentryId();
+            trace.SpanId = activity.SpanId.AsSentrySpanId();
+        }
+
+        return transaction;
+    }
+
+    /// <inheritdoc />
+    public override void OnStart(Activity data)
+    {
+        if (data.ParentSpanId != default && _map.TryGetValue(data.ParentSpanId, out var parentSpan))
+        {
+            // We can find the parent span - start a child span.
+            var context = new SpanContext(
+                data.SpanId.AsSentrySpanId(),
+                data.ParentSpanId.AsSentrySpanId(),
+                data.TraceId.AsSentryId(),
+                data.OperationName,
+                data.DisplayName,
+                null,
+                null)
+            {
+                Instrumenter = Instrumenter.OpenTelemetry
+            };
+
+            var span = (SpanTracer)parentSpan.StartChild(context);
+            span.StartTimestamp = data.StartTimeUtc;
+            _map[data.SpanId] = span;
+        }
+        else
+        {
+            // If a parent exists at all, then copy its sampling decision.
+            bool? isSampled = data.HasRemoteParent ? data.Recorded : null;
+
+            // No parent span found - start a new transaction
+            var transactionContext = new TransactionContext(
+                data.SpanId.AsSentrySpanId(),
+                data.ParentSpanId.AsSentrySpanId(),
+                data.TraceId.AsSentryId(),
+                data.DisplayName,
+                data.OperationName,
+                data.DisplayName,
+                null,
+                isSampled,
+                isSampled)
+            {
+                Instrumenter = Instrumenter.OpenTelemetry
+            };
+
+            var baggageHeader = BaggageHeader.Create(data.Baggage.WithValues());
+            var dynamicSamplingContext = baggageHeader.CreateDynamicSamplingContext();
+            var transaction = (TransactionTracer)_hub.StartTransaction(
+                transactionContext, new Dictionary<string, object?>(), dynamicSamplingContext
+                );
+            transaction.StartTimestamp = data.StartTimeUtc;
+            _map[data.SpanId] = transaction;
+        }
+    }
+
+    /// <inheritdoc />
+    public override void OnEnd(Activity data)
+    {
+        // Make a dictionary of the attributes (aka "tags") for faster lookup when used throughout the processor.
+        var attributes = data.TagObjects.ToDictionary();
+
+        if (attributes.TryGetTypedValue("http.url", out string url) && _hub.IsSentryRequest(url))
```

Comment:
Since the method was used in both `SentrySpanProcessor` and `SentryPropagator` I opted to keep it separate from both of those classes as an [extension method on SentryOptions](https://github.com/getsentry/sentry-dotnet/blob/f85fb9d9c96eed7c6e99f4aa11433be86d78c2fc/src/Sentry.OpenTelemetry/SentryOptionsExtensions.cs#L20).

That does tidy things up a bit. It gets rid of the explicit interface implementation and the `new Uri()`.

If we really wanted to, we could make it a plain vanilla method on the SentryOptions class... for the time being, the only place it's used is in the OpenTelemetry package so I reckon we keep it here and promote it to sit on SentryOptions if/when it's needed beyond implementing OpenTelemetry.